invokeai/backend/image_util/infill_methods/lama.py

```diff
@@ -28,18 +25,14 @@ def load_jit_model(url_or_path, device):
 
 
 class LaMA:
-    def __call__(self, input_image: Image.Image, *args: Any, **kwds: Any) -> Any:
-        device = TorchDevice.choose_torch_device()
-        model_location = get_config().models_path / "core/misc/lama/lama.pt"
-
-        if not model_location.exists():
-            download_with_progress_bar(
-                name="LaMa Inpainting Model",
-                url="https://github.com/Sanster/models/releases/download/add_big_lama/big-lama.pt",
-                dest_path=model_location,
-            )
+    def __init__(self, context: InvocationContext):
```

Comment:
We shouldn't be passing the context into these classes, it makes them unusable outside nodes.

Reply:
Ok. I'll have to refactor `invocation_context.py` and move `load_ckpt_from_url()` into the model manager service. It probably belongs there anyway. Getting late, so leaving till tomorrow.

Reply:
Done now, but I had to figure out a way to pass the currently active model manager service to the LaMA constructor, and ended up doing this in `invocations/infill.py`:

```
lama = LaMA(context._services.model_manager)
```

I'd prefer not to access the protected `_services` attribute. Is there a better way to do this?